File: backend/app/schemas/course.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CourseBase(BaseModel):
    title: str = Field(..., min_length=3, max_length=180)
    description: str = Field(..., min_length=1)


class CourseRead(CourseBase):
    id: int
    filename: str
    original_filename: str
    file_path: str
    file_size: int
    created_at: datetime
    updated_at: datetime
    pdf_url: str

    model_config = ConfigDict(from_attributes=True)


class AdminStats(BaseModel):
    total_courses: int
    total_pdf_files: int
    total_storage_bytes: int
    latest_course: CourseRead | None

File: frontend/src/types/course.ts
export type Course = {
  id: number;
  title: string;
  description: string;
  filename: string;
  original_filename: string;
  file_path: string;
  file_size: number;
  created_at: string;
  updated_at: string;
  pdf_url: string;
};

export type AdminStats = {
  total_courses: number;
  total_pdf_files: number;
  total_storage_bytes: number;
  latest_course: Course | null;
};

export type CourseFormValues = {
  title: string;
  description: string;
  file?: File | null;
};
